ludwig/schema/trainer.py

```diff
@@ -188,5 +217,250 @@ class TrainerConfig(schema_utils.BaseMarshmallowConfig):
+    )
+
+    max_cat_threshold: int = schema_utils.PositiveInteger(
+        default=32,
+        description="Number of split points considered for categorical features for the GBM trainer.",
+    )
+
+    cat_l2: float = schema_utils.NonNegativeFloat(
+        default=10.0, description="L2 regularization factor for categorical split in the GBM trainer."
+    )
+
+    cat_smooth: float = schema_utils.NonNegativeFloat(
+        default=10.0, description="Smoothing factor for categorical split in the GBM trainer."
+    )
+
+    max_cat_to_onehot: int = schema_utils.PositiveInteger(
+        default=4,
+        description="Maximum categorical cardinality required before one-hot encoding in the GBM trainer.",
+    )
+
+    cegb_tradeoff: float = schema_utils.NonNegativeFloat(
+        default=1.0,
+        description="Cost-effective gradient boosting multiplier for all penalties in the GBM trainer.",
+    )
+
+    cegb_penalty_split: float = schema_utils.NonNegativeFloat(
+        default=0.0,
+        description="Cost-effective gradient boosting penalty for splitting a node in the GBM trainer.",
+    )
+
+    path_smooth: float = schema_utils.NonNegativeFloat(
+        default=0.0,
+        description="Smoothing factor applied to tree nodes in the GBM trainer.",
+    )
+
+    verbose: int = schema_utils.IntegerRange(default=0, min=-1, max=2, description="Verbosity level for GBM trainer.")
+
+    # LightGBM IO params
+    max_bin: int = schema_utils.PositiveInteger(
+        default=255, description="Maximum number of bins to use for discretizing features with GBM trainer."
+    )
+
+
+def get_model_type_jsonschema():
+    return {"type": "string", "enum": [MODEL_ECD, MODEL_GBM], "default": MODEL_ECD}
+
+
 def get_trainer_jsonschema():
-    return schema_utils.unload_jsonschema_from_marshmallow_class(TrainerConfig)
+    def allowed_types_for_trainer_schema(cls) -> List[str]:
+        """Returns the allowed values for the "type" field on the given trainer schema."""
+        return cls.Schema().fields[TYPE].validate.choices
+
+    conds = []
+    all_trainer_types = []
+    for trainer in trainer_schema_registry:
+        trainer_cls = trainer_schema_registry[trainer]
+
+        allowed_trainer_types = allowed_types_for_trainer_schema(trainer_cls)
+        all_trainer_types.extend(allowed_trainer_types)
+
+        other_props = schema_utils.unload_jsonschema_from_marshmallow_class(trainer_cls)["properties"]
+        other_props.pop("type")
+        for trainer_type in allowed_trainer_types:
+            trainer_cond = schema_utils.create_cond(
+                {"type": trainer_type},
+                other_props,
+            )
+            conds.append(trainer_cond)
+
+    return {
+        "type": "object",
+        "properties": {
+            "type": {"type": "string", "enum": all_trainer_types},
+        },
+        "required": ["type"],
+        "title": "trainer_options",
+        "allOf": conds,
+        "description": "",
```

Comment:
Please add a description :) - in this case something brief to describe the different enum choices and how they interact with `model_type`.